cms/djangoapps/maintenance/views.py

```diff
@@ -0,0 +1,217 @@
+"""
+Views for the maintenance app.
+"""
+import logging
+from django.db import transaction
+from django.core.urlresolvers import reverse_lazy
+from django.utils.decorators import method_decorator
+from django.utils.translation import ugettext_lazy as _
+from django.views.generic import View
+
+from edxmako.shortcuts import render_to_response
+from opaque_keys import InvalidKeyError
+from opaque_keys.edx.keys import CourseKey
+from xmodule.modulestore.django import modulestore
+from xmodule.modulestore.exceptions import ItemNotFoundError
+
+from contentstore.management.commands.utils import get_course_versions
+from util.views import require_global_staff
+
+
+log = logging.getLogger(__name__)
+
+MAINTENANCE_COMMANDS = {
```

Comment:
You can still get all that if you define these on the index view.
in sudo-python:
```python
def get_urls():
    urls = []
    for view in IndexView.views:
        urls.append(view.url)
```

or something like that.

These attributes should be defined on each class-based view